sql/sql_select.cc

```diff
@@ -20142,6 +20200,272 @@ simplify_joins(JOIN *join, List<TABLE_LIST> *join_list, COND *conds, bool top,
 }
 
 
+/**
+   Rewrite a FULL JOIN to a LEFT JOIN by mutating the
+   left and right table state to make them appear as though
+   the user wrote the FULL JOIN as a LEFT JOIN originally.
+
+   @param left_table  table t1 in t1 FULL JOIN t2
+   @param right_table table t2 in t1 FULL JOIN t2
+*/
+
+static void rewrite_full_to_left(TABLE_LIST *left_table,
+                                 TABLE_LIST *right_table)
+{
+  // Grammar does not mark the left table at all
+  left_table->outer_join= 0;
+
+  /*
+    Clear FULL JOIN flag and do as the grammar does by marking
+    the right table as JOIN_TYPE_LEFT.
+  */
+  right_table->outer_join= JOIN_TYPE_LEFT;
+
+  /*
+    The right table must have an ON clause.  NATURAL JOINs get
+    this not from the grammar but they're built before simplify_joins
+    is called.
+  */
+  DBUG_ASSERT(right_table->on_expr);
+
+  // Only the right table in a LEFT JOIN has the naming context in the grammar
+  left_table->on_context= nullptr;
+
+  // The grammar 'search_condition: ' rule marks this.
```

Comment:
So, an Item* object has been a FULL OUTER JOIN's ON expression and didn't have IS_COND flag?
But now it becomes a LEFT JOIN's ON expression and we're setting the flag.

A Naive question is, why don't we set IS_COND flag for FULL OUTER JOINs' ON expressions?